code/game/objects/items/flamethrower.dm

```diff
@@ -214,13 +214,22 @@
 	//TODO: DEFERRED Consider checking to make sure tank pressure is high enough before doing this...
 	//Transfer 5% of current tank air contents to turf
 	var/datum/gas_mixture/air_transfer = ptank.air_contents.remove_ratio(release_amount)
-	if(air_transfer.gases[/datum/gas/plasma])
-		air_transfer.gases[/datum/gas/plasma][MOLES] *= 5
+	if(air_transfer.gases[/datum/gas/plasma] || air_transfer.gases[/datum/gas/tritium] || air_transfer.gases[/datum/gas/hydrogen])
+		if(air_transfer.gases[/datum/gas/plasma])
+			air_transfer.gases[/datum/gas/plasma][MOLES] *= 5
+		if(air_transfer.gases[/datum/gas/tritium])
+			air_transfer.gases[/datum/gas/tritium][MOLES] *= 6.5
+		if(air_transfer.gases[/datum/gas/hydrogen])
+			air_transfer.gases[/datum/gas/hydrogen][MOLES] *= 7
+	else
+		air_transfer.temperature += 15000
 	target.assume_air(air_transfer)
 	//Burn it based on transfered gas
 	target.hotspot_expose((ptank.air_contents.temperature*2) + 380,500)
 	//location.hotspot_expose(1000,500,1)
 	SSair.add_to_active(target, 0)
+	target.air_update_turf()
```

Comment:
You don't need the update_turf here as add to active is already being called

Reply:
gotcha